core/src/main/java/com/epam/esm/dao/CertificateJdbcDao.java

```diff
@@ -0,0 +1,154 @@
+                    " WHERE name LIKE ? ORDER BY gift_certificate.id";
+    private static final String DELETE_BY_ID =
+            "DELETE FROM gift_certificate WHERE id = ?";
+    private static final String FIND_CERTIFICATE_TAG =
+            "SELECT COUNT(*) FROM certificate_tag WHERE certificate_id = ? AND tag_id = ?";
+
+    private static final String ID = "ID";
+    private final CertificateResultSetExtractor certificateResultSetExtractor;
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public CertificateJdbcDao(CertificateResultSetExtractor certificateResultSetExtractor, JdbcTemplate jdbcTemplate) {
+        this.certificateResultSetExtractor = certificateResultSetExtractor;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public List<Certificate> getAll() {
+        return jdbcTemplate.query(GET_ALL, certificateResultSetExtractor);
+    }
+
+    @Override
+    public List<Certificate> getByTagName(String tagName) {
+        return jdbcTemplate.query(GET_BY_TAG_NAME, certificateResultSetExtractor, tagName);
+    }
+
+    @Override
+    public List<Certificate> getByNamePart(String namePart) {
+        String formattedNamePart = "%" + namePart + "%";
+        return jdbcTemplate.query(GET_BY_NAME_PART, certificateResultSetExtractor, formattedNamePart);
+    }
+
+    @Override
+    public Optional<Certificate> getById(long id) {
+        List<Certificate> query = jdbcTemplate.query(GET_BY_ID, certificateResultSetExtractor, id);
+        if (query.size() > 1) {
+            throw new MultipleRecordsWereFoundException(MULTIPLE_RECORDS_WERE_FOUND_BY_ID, id);
+        }
+        return query.stream().findAny();
+    }
+
+    @Override
+    public void update(Certificate entity) {
+        LocalDateTime createDate = entity.getCreateDate();
+        LocalDateTime lastUpdateDate = entity.getLastUpdateDate();
+        jdbcTemplate.update(UPDATE_CERTIFICATE,
+                entity.getName(),
+                entity.getDescription(),
+                entity.getPrice(),
+                entity.getDuration(),
+                createDate != null ? createDate.toString() : null,
+                lastUpdateDate != null ? lastUpdateDate.toString() : null,
+                entity.getId());
+    }
+
+    @Override
+    public long create(Certificate entity) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(connection -> {
+            PreparedStatement preparedStatement = connection.prepareStatement(INSERT_CERTIFICATE, new String[]{ID});
+            preparedStatement.setString(1, entity.getName());
+            preparedStatement.setString(2, entity.getDescription());
+            preparedStatement.setInt(3, entity.getPrice());
+            preparedStatement.setInt(4, entity.getDuration());
+            preparedStatement.setString(5, entity.getCreateDate() != null ?
```

Comment:
@KukharchukDaniil Not necessary check this fields. Entity for create should be prepared on service layer.

Reply:
fixed